cmd/kodelet/gha_agent_onboard.go

```diff
@@ -0,0 +1,339 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// GhaAgentOnboardConfig holds configuration for the gha-agent-onboard command
+type GhaAgentOnboardConfig struct {
+	GithubApp             string
+	AuthGatewayEndpoint   string
+}
+
+// NewGhaAgentOnboardConfig creates a new GhaAgentOnboardConfig with default values
+func NewGhaAgentOnboardConfig() *GhaAgentOnboardConfig {
+	return &GhaAgentOnboardConfig{
+		GithubApp:           "kodelet",
+		AuthGatewayEndpoint: "https://gha-auth-gateway.kodelet.com/api/github",
+	}
+}
+
+// GitHub workflow template
+const workflowTemplate = `name: Background Kodelet
+
+on:
+  issue_comment:
+    types: [created]
+  issues:
+    types: [opened, assigned]
+  pull_request_review_comment:
+    types: [created]
+  pull_request_review:
+    types: [submitted]
+
+env:
+  TIMEOUT_MINUTES: "300"
+
+jobs:
+  background-kodelet:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      issues: read
+      pull-requests: read
+      contents: read
+    timeout-minutes: 15  # 15 Minutes
+    if: |
+      (
+        (github.event_name == 'issues' && contains(github.event.issue.body, '@kodelet')) ||
+        (github.event_name == 'issue_comment' && contains(github.event.comment.body, '@kodelet')) ||
+        (github.event_name == 'pull_request_review_comment' && contains(github.event.comment.body, '@kodelet')) ||
+        (github.event_name == 'pull_request_review' && contains(github.event.review.body, '@kodelet'))
+      ) &&
+      (
+        (github.event.issue.author_association == 'OWNER' || github.event.issue.author_association == 'MEMBER' || github.event.issue.author_association == 'COLLABORATOR') ||
+        (github.event.comment.author_association == 'OWNER' || github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'COLLABORATOR') ||
+        (github.event.review.author_association == 'OWNER' || github.event.review.author_association == 'MEMBER' || github.event.review.author_association == 'COLLABORATOR')
+      )
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Agent Environment
+        run: |
+          echo "YMMV"
+      - name: Run Kodelet
+        uses: jingkaihe/kodelet-action@v0.1.7-alpha
+        with:
+          anthropic-api-key: ${{ secrets.ANTHROPIC_API_KEY }}
```

Comment:
add `auth-gateway-endpoint: $gateway_endpoint` and replace the endpoint with `AuthGatewayEndpoint` from the cli